client/network/src/protocol/sync.rs

```diff
@@ -192,6 +207,12 @@ pub struct ChainSync<B: BlockT> {
 	max_parallel_downloads: u32,
 	/// Total number of downloaded blocks.
 	downloaded_blocks: usize,
+	/// All block announcement that are currently being validated.
+	block_announce_validation: FuturesUnordered<
+		Pin<Box<dyn Future<Output = PreValidateBlockAnnounce<B::Header>> + Send>>
+	>,
+	/// Stats per peer about the number of concurrent block announce validations.
+	block_announce_validation_per_peer_stats: HashMap<PeerId, usize>,
```

Comment:
This needs to be cleaned up when peer disconnects

Reply:
Why? The validations are still running and when they are finished, we will clean this hash map